src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitService.java

```diff
@@ -47,8 +49,36 @@ public UUID runAndSaveResult(ShortCircuitRunContext runContext) {
         return resultUuid;
     }
 
-    public String getResult(UUID resultUuid) {
-        return resultRepository.find(resultUuid);
+    private static ShortCircuitAnalysisResult fromEntity(ShortCircuitAnalysisResultEntity resultEntity, boolean full) {
+        List<FaultResult> faultResults = resultEntity.getFaultResults().stream().filter(fr -> full || !fr.getLimitViolations().isEmpty()).map(fr -> fromEntity(fr)).collect(Collectors.toList());
+        return new ShortCircuitAnalysisResult(resultEntity.getResultUuid(), resultEntity.getWriteTimeStamp(), faultResults);
+    }
+
+    private static FaultResult fromEntity(FaultResultEntity faultResultEntity) {
+        Fault fault = fromEntity(faultResultEntity.getFault());
+        double current = faultResultEntity.getCurrent();
+        double shortCircuitPower = faultResultEntity.getShortCircuitPower();
+        List<LimitViolation> limitViolations = faultResultEntity.getLimitViolations().stream().map(lv -> fromEntity(lv)).collect(Collectors.toList());
+        List<FeederResult> feederResults = faultResultEntity.getFeederResults().stream().map(fr -> fromEntity(fr)).collect(Collectors.toList());
+        return new FaultResult(fault, current, shortCircuitPower, limitViolations, feederResults);
+    }
+
+    private static Fault fromEntity(FaultEmbeddable faultEmbeddable) {
+        return new Fault(faultEmbeddable.getId(), faultEmbeddable.getElementId(), faultEmbeddable.getFaultType().name());
+    }
+
+    private static LimitViolation fromEntity(LimitViolationEmbeddable limitViolationEmbeddable) {
+        return new LimitViolation(limitViolationEmbeddable.getSubjectId(), limitViolationEmbeddable.getLimitType().name(),
+                limitViolationEmbeddable.getLimit(), limitViolationEmbeddable.getLimitName(), limitViolationEmbeddable.getValue());
+    }
+
+    private static FeederResult fromEntity(FeederResultEmbeddable feederResultEmbeddable) {
+        return new FeederResult(feederResultEmbeddable.getConnectableId(), feederResultEmbeddable.getCurrent());
+    }
+
+    public ShortCircuitAnalysisResult getResult(UUID resultUuid, boolean full) {
+        ShortCircuitAnalysisResultEntity result = resultRepository.find(resultUuid);
+        return result != null ? fromEntity(result, full) : null;
```

Comment:
Maybe if we return an Optional in the repository all the chain will be cleaner ?

Reply:
Why not, modified.